[Lin Yuan Xun, Caleb] iP

Caleblyx opened the pull request.

Commit:
Add Gradle support

Caleblyx committed:
Level1 - Greet, Echo, Exit

Caleblyx committed:
Level 2. Add, List

Caleblyx committed:
Level-3: Mark as Done

Caleblyx committed:
Level-4: ToDo, Event, Deadline

Caleblyx committed:
A-TextUiTesting: Automated Text UI Testing

Caleblyx committed:
Level-5: Handle Errors

Caleblyx committed:
Level-6: Delete

Caleblyx committed:
Done exception

Caleblyx committed:
Level 7. Save

Caleblyx committed:
Level 8. Dates and Times

Caleblyx committed:
Merge branch 'branch-level-7'

Caleblyx committed:
Merge branch 'branch-level-8'

# Conflicts:
#	src/main/java/Bob.java

Resolution: keep all changes

Caleblyx committed:
A-MoreOOP

Caleblyx committed:
A-JUnit

Add J-Unit tests for the following classes: Parser, TaskList, and Event.

Caleblyx committed:
A-JavaDoc

Caleblyx committed:
Add JavaDoc for class variables of Storage class

Caleblyx committed:
branch-A-CodingStandard

Improve coding standard

Caleblyx committed:
branch-level-9

Caleblyx committed:
Merge tag 'branch-A-JavaDoc'

Caleblyx committed:
Merge tag 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/AddCommand.java
#	src/main/java/Bob.java
#	src/main/java/BobIncompleteEventDescriptionException.java
#	src/main/java/Deadline.java
#	src/main/java/DeleteCommand.java
#	src/main/java/Event.java
#	src/main/java/ExitCommand.java
#	src/main/java/Storage.java
#	src/main/java/Task.java
#	src/main/java/TaskList.java
#	src/main/java/Todo.java
#	src/main/java/UI.java

Resolution: Keep javadocs and coding standard improvements

Caleblyx committed:
Merge branch 'branch-level-9'

# Conflicts:
#	data/save.txt
#	src/main/java/TaskList.java

Resolution: keep all changes made in the two branches.

aqcd commented on data/save.txt:
```diff
@@ -0,0 +1,10 @@
+<<<<<<< HEAD
```
This might have been missed while merging! Should this be here?

aqcd commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,41 @@
+import main.java.*;
```
Would it be clearer to list the imported classes explicitly?

aqcd commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,41 @@
+public class Bob {
+    private TaskList tasks = new TaskList();
+    private Storage storage;
+    private UI uI;
```
Should this variable be named "ui" instead?

aqcd commented on src/main/java/BobDateTimeParseException.java:
```diff
@@ -0,0 +1,20 @@
+/**
+ * An exception that may be thrown when an error occurs during the parsing of dates and times.
+ */
+public class BobDateTimeParseException extends BobException {
```
I like that you made classes for the specific types of exceptions!

aqcd commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,113 @@
+     */
+    public Event (String description, String period) {
+        super(description);
+        this.start = LocalDateTime.parse(period.substring(0,15), inputFormatter);
```
Should the call to .substring be formatted in a better way? Noticed this in a few other places as well!

aqcd commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,113 @@
+     */
+    @Override
+    public String toString() {
+        return "[" + getStatusIcon() + "] " + getDescription() + " (at: " + getPeriod() +")";
```
Should there be better spacing with the '+'s?

aqcd commented on src/main/java/ExitCommand.java:
```diff
@@ -0,0 +1,29 @@
+public class ExitCommand extends Command {
+
+    /**
+     * Executes nothing.
```
Should there be an explanation on why this method executes nothing? I expected it to be the method that terminates the application.

aqcd commented on src/main/java/ExitCommand.java:
```diff
@@ -0,0 +1,29 @@
+    /**
+     * Returns false to indicate that Bob may terminate.
+     *
+     * @return false.
```
Should this method be returning false? It's currently returning true.

aqcd commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,170 @@
+        }
+    }
+
+    /**Updates the Storage's text file according to the tasks in a provided TaskList.
```
Should this JavaDoc be in the next line?

aqcd commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,170 @@
+
+    /**
+     * Closes the Storage's writer.
+     * @throws BobIOException if the Storage's text file does not exist.
```
Should this line be 1 extra line below the previous? Noticed this in some other places too!

aqcd commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,27 @@
+     * @param isDone indicates if the todo is completed.
+     * @param description brief description of the todo.
+     */
+
```
Noticed some extra lines after JavaDocs. Should the JavaDoc style be consistent?

aqcd commented on src/main/java/UI.java:
```diff
@@ -0,0 +1,112 @@
+     * @throws BobIndexOutOfBoundsException if the method tries to retrieve a task with an index not on the list.
+     */
+    public void printOutList(TaskList tasks) throws BobIndexOutOfBoundsException {
+        for(int i = 1; i < tasks.size()+1; i++) {
```
Should this line be better spaced?

aqcd reviewed (commented):
The main functionality looks good!
Some parts of the code, such as the Parser class, could do with more documentation as I was rather confused as to what the code is doing.
A few cosmetic changes requested.

Caleblyx committed:
add /src/test/.idea to .gitignore

Caleblyx committed:
Merge branch 'add-gradle-support'

Caleblyx committed:
Set up Gradle to build, run, and test using Gradle

kerkpy commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,64 @@
+     * @return true or false if AddCommand is equal or not equal to the object respectively.
+     */
+    @Override
+    public boolean equals(Object o) {
```
Could the variables for be more descriptive? That is, instead of o, be obj, and instead of c, be cmd

kerkpy commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,107 @@
+ */
+public class Task {
+    /** A brief description of the task. */
+    protected String description;
```
Should they have been private instead of protected?

kerkpy commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+public class TaskList {
+
+    /** An array list consisting of tasks. */
+    private ArrayList<Task> list;
```
should this have been renamed as tasks?

kerkpy commented on src/main/java/UI.java:
```diff
@@ -0,0 +1,112 @@
+    public void printOutList(TaskList tasks) throws BobIndexOutOfBoundsException {
+        for(int i = 1; i < tasks.size()+1; i++) {
+            Task task = tasks.get(i);
+            System.out.println(i +"." + task.toString());
```
Should there be spacing between +"." ?

kerkpy commented on src/main/java/UI.java:
```diff
@@ -0,0 +1,112 @@
+            Task task = tasks.get(i);
+            System.out.println(i +"." + task.toString());
+        }
+        if(tasks.isEmpty()) {
```
Should there be spacing between if(..)?

kerkpy reviewed (commented):
The coding style looks good!

Caleblyx committed:
branch-level-10

add a GUI

Caleblyx committed:
Merge tag 'branch-level-10'

Caleblyx committed:
branch-A-CheckStyle

Caleblyx committed:
Merge branch 'branch-A-CheckStyle'

Caleblyx committed:
Improve code quality

Caleblyx committed:
add assertions to code

Caleblyx committed:
Make further improvements to code quality of Parser.java

Caleblyx committed:
Merge pull request #1 from Caleblyx/heads/branch-A-Assertions

add assertions to code

Caleblyx committed:
Merge tag 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/bob/command/DeleteCommand.java

Resolution:
Retain changes made by both branches

Caleblyx committed:
Add snooze and reschedule commands

Caleblyx committed:
fix minor syntax errors

Caleblyx committed:
add exception handling for snoozing, rescheduling, and event or deadline creation

Caleblyx committed:
Merge branch 'branch-BCD-extension-snooze'

Caleblyx committed:
improve GUI

Caleblyx committed:
make profile picture circular

Caleblyx committed:
make dialogbox resize dynamically according to content

Caleblyx committed:
change format of error messages on GUI

Caleblyx committed:
Improve GUI

Change Bob profile picture
Improve label design and font
Improve error message GUI

Caleblyx committed:
Modify exception handling

-Change exception messages
-Edit exception handling of Parse.parseDelete and Parse.parseDone

Caleblyx committed:
Improve delete and done commands

-Add option to delete all tasks
-Add option to mark all tasks as done

Caleblyx committed:
Rename exception class

Caleblyx committed:
Modify javadocs

-Add javadocs for new methods.
-Edit old javadocs

Caleblyx committed:
Set theme jekyll-theme-cayman

Caleblyx committed:
Add credits for adapted code

Caleblyx committed:
Delete _config.yml

Caleblyx committed:
Merge branch 'master' of https://github.com/Caleblyx/ip

Caleblyx committed:
Edit introduction README.MD

Caleblyx committed:
Add user guide and edit portfolio page

Caleblyx committed:
Modify portfolio page

Caleblyx committed:
Change User Guide URL on portfolio page

Caleblyx committed:
Modify portfolio page

Caleblyx committed:
Modify build.gradle file
